src/vibrator-sysfs.cpp

```diff
@@ -18,27 +18,65 @@
 
 #include "vibrator-sysfs.h"
 
+#include <fstream>
+#include <string>
 #include <iostream>
 #include <unistd.h>
 
 // if the kernel support led triggers
 // we set 
 namespace hfd {
 
+// True iff the standard transient trigger is supported.
+// Some devices implement the transient trigger but don't bother
+// Actually support it, this function will return a false negative
+// in that case
+bool VibratorSysfs::supportTransient() {
+    std::ifstream source_stream("/sys/class/leds/vibrator/trigger");
+
+    if (!source_stream) {
+        std::cerr << "Can't read '/sys/class/leds/vibrator/trigger', does the file exist?" << std::endl;
+        return false;
+    }
+
+    std::string supported_triggers = std::string(std::istreambuf_iterator<char>(source_stream),
+                                std::istreambuf_iterator<char>());
+    
+    bool transientSupported = supported_triggers.find("transient") != std::string::npos;
+    std::cout << "Vibrator supports transient trigger: " << transientSupported << std::endl;
+
+    return transientSupported;
+}
+
+// Checks that the transient trigger is supported,
+// or implemented manually by the driver
 bool VibratorSysfs::usable() {
-    return access("/sys/class/leds/vibrator", F_OK ) != -1;
+    // Need to be able to access the vibrator
+    bool usable = (access("/sys/class/leds/vibrator", F_OK ) != -1);
```

Comment:
Early return if this actually fails?

Reply:
Thinking of it again, vibrator is usable by sysfs plugin if the sysfs path and duration/state/activate exist, no need to even check if it supports transient.

Reply:
I considered that, however from what I can tell the `duration/state/activate` properties may not appear until transient trigger is enabled, removing the check for it would mean that device which DO properly implement the transient trigger would fail the check.